jfrog-cli/artifactory/utils/nuget/solution/solution.go

```diff
@@ -62,27 +67,88 @@ func (solution *solution) GetProjects() []project.Project {
 }
 
 func (solution *solution) loadProjects() error {
-	return filepath.Walk(solution.path, func(path string, f os.FileInfo, err error) error {
+	regExp, err := utils.GetRegExp(`Project\("(.*)\nEndProject`)
+	if err != nil {
+		return err
+	}
+
+	allProjects, err := solution.getProjectsFromSlns(regExp)
+	if err != nil {
+		return err
+	}
+
+	for _, projectLine := range allProjects {
+		projectName, csprojPath, err := parseProject(projectLine, solution.path)
 		if err != nil {
-			return errorutils.CheckError(err)
+			log.Error(err)
+			continue
 		}
-		if f.IsDir() {
-			return nil
+		proj, err := project.Load(projectName, filepath.Dir(csprojPath), csprojPath)
+		if err != nil {
+			log.Error(err)
+			continue
 		}
-		if filepath.Ext(f.Name()) == ".csproj" {
-			projectName := strings.TrimSuffix(f.Name(), ".csproj")
-			csprojPath, err := filepath.Rel(solution.path, path)
-			if err != nil {
-				return errorutils.CheckError(err)
-			}
-			proj, err := project.Load(projectName, filepath.Dir(csprojPath), csprojPath)
+		if proj.Extractor() != nil {
+			solution.projects = append(solution.projects, proj)
+		}
+	}
+	return nil
+}
+
+// Finds all the project section within the sln files. If sln file is not provided,
+// searching the entire directory for all sln files.
+// Returns slice of all the projects.
+func (solution *solution) getProjectsFromSlns(regExp *regexp.Regexp) ([]string, error) {
+	var allProjects []string
+	if solution.slnFile == "" {
+		slnFiles, err := fileutils.ListFilesWithSpecificExtension(solution.path, ".sln")
+		if err != nil {
+			return nil, err
+		}
+		for _, slnFile := range slnFiles {
+			projects, err := parseSlnFile(slnFile, regExp)
 			if err != nil {
-				return err
-			}
-			if proj.Extractor() != nil {
-				solution.projects = append(solution.projects, proj)
+				return nil, err
 			}
+			allProjects = append(allProjects, projects...)
 		}
-		return nil
-	})
+	} else {
+		projects, err := parseSlnFile(filepath.Join(solution.path, solution.slnFile), regExp)
+		if err != nil {
+			return nil, err
+		}
+		allProjects = append(allProjects, projects...)
+	}
+	return allProjects, nil
+}
+
+// Parsing the project line for the project name and path information.
+// Returns the name and path to csproj
+func parseProject(projectLine, path string) (projectName, csprojPath string, err error) {
+	parsedLine := strings.Split(projectLine, "=")
+	if len(parsedLine) <= 1 {
+		return "", "", errors.New("Wrong number of arguments for project line:" + projectLine)
+	}
+
+	projectInfo := strings.Split(parsedLine[1], ",")
+	if len(projectInfo) <= 2 {
+		return "", "", errors.New("Wrong number of arguments for project information: " + parsedLine[1])
```

Comment:
Suggesting to change:
Wrong number of arguments for project information:
to:
Unexpected project information format:

Reply:
Done